src/ape_ethereum/ecosystem.py

```diff
@@ -442,13 +443,54 @@ def decode_logs(
                 continue
 
             event_arguments = abi.decode(topics, log["data"])
-
             yield ContractLog(
-                name=abi.event_name,
+                block_hash=log["blockHash"],
+                block_number=to_int(log["blockNumber"]),
                 contract_address=self.decode_address(log["address"]),
                 event_arguments=event_arguments,
+                log_index=log["logIndex"],
+                name=abi.event_name,
                 transaction_hash=log["transactionHash"],
-                block_number=to_int(log["blockNumber"]),
-                block_hash=log["blockHash"],
-                log_index=to_int(log["logIndex"]),
-            )
+            )  # type: ignore
+
+    def decode_library_log(self, log: Dict) -> Optional[ContractLog]:
+        return self._decode_ds_note(log)
+
+    def _decode_ds_note(self, log: Dict) -> Optional[ContractLog]:
```

Comment:
Could this function live under `ape_ethereum.transactions.Receipt` instead, and be used directly instead of the indirection involved w/ adding `EcosystemAPI.decode_library_log`. It doesn't seem to be used by other ecosystems, so I think it would be more at home reducing the overall complexity of the API layer a bit.

Reply:
Would that limit the use-case of decoding DS Notes outside the context of having a receipt?
Like what if I got the data from someplace else and I want to decode it, I could use provider method alone by having it this way.

Reply:
You can still use the method, it's just private. I don't have a problem with it being a public method either, just don't think it should be an API method